core/src/test/java/org/elasticsearch/search/profile/query/CollectorResultTests.java

```diff
@@ -22,15 +22,55 @@
 import org.elasticsearch.common.xcontent.ToXContent;
 import org.elasticsearch.common.xcontent.XContentBuilder;
 import org.elasticsearch.common.xcontent.XContentFactory;
+import org.elasticsearch.common.xcontent.XContentParser;
+import org.elasticsearch.common.xcontent.XContentType;
 import org.elasticsearch.test.ESTestCase;
 
 import java.io.IOException;
 import java.util.ArrayList;
 import java.util.Collections;
 import java.util.List;
 
+import static org.elasticsearch.common.xcontent.XContentHelper.toXContent;
+import static org.elasticsearch.common.xcontent.XContentParserUtils.ensureExpectedToken;
+import static org.elasticsearch.test.hamcrest.ElasticsearchAssertions.assertToXContentEquivalent;
+
 public class CollectorResultTests extends ESTestCase {
 
+    public static CollectorResult createTestItem(int depth) {
+        String name = randomAsciiOfLengthBetween(5, 10);
+        String reason = randomAsciiOfLengthBetween(5, 10);
+        long time = randomNonNegativeLong();
+        if (randomBoolean()) {
+            // also often use relatively "small" values, otherwise we will mostly test huge longs
+            time = time % 100000;
+        }
+        int size = randomIntBetween(0, 5);
+        List<CollectorResult> children = new ArrayList<>(size);
+        if (depth > 0) {
+            for (int i = 0; i < size; i++) {
+                children.add(createTestItem(depth - 1));
+            }
+        }
+        return new CollectorResult(name, reason, time, children);
+    }
+
+
+    public void testFromXContent() throws IOException {
+        CollectorResult collectorResult = createTestItem(1);
+        XContentType xcontentType = randomFrom(XContentType.values());
+        XContentBuilder builder = XContentFactory.contentBuilder(xcontentType);
+        boolean humanReadable = randomBoolean();
```

Comment:
I meant `builder.humanReadable(randomBoolean());` here and use `builder.humanReadable() ` later